git/git.go

```diff
@@ -714,6 +714,63 @@ func CloneWithoutFilters(args []string) error {
 	return nil
 }
 
+// RemoteRefs returns a list of branches & tags for a remote
+// If retrieveFromRemote is true, this information is built by actually calling
+// the remote using 'git ls-remote'. Otherwise the cached remote branch
+// information held locally is used instead.
+func RemoteRefs(remoteName string, retrieveFromRemote bool) ([]*Ref, error) {
+
```

Comment:
I think I'd rather have two separate functions, `RemoteRefs()` and `CachedRemoteRefs()`. There's not much shared between the two commands, and switching between the two values based on a boolean arg feels weird.

I'd also prefer the scanner loop use `continue`, instead of continuing to nest `if` statements. It results in a few more lines, but less indentation, which is preferable to me personally.

```golang
func RemoteRefs() {
	// init command, build scanner, etc
	for scanner.Scan() {
		match := r.FindStringSubmatch(scanner.Text())
		if match == nil {
			continue
		}
		
		name := strings.TrimSpace(match[3])
		// Don't match head
		if name == "HEAD" {
			continue
		}
		
		sha := match[1]
		if match[2] == "heads" {
			ret = append(ret, &Ref{name, RefTypeRemoteBranch, sha})
		} else {
			ret = append(ret, &Ref{name, RefTypeRemoteTag, sha})
		}
	}
}
```